autopush/base.py

```diff
@@ -45,3 +45,33 @@ def write_error(self, code, **kwargs):
             self.log.failure("Error in handler: %s" % code,
                              client_info=self._client_info)
         self.finish()
+
+    def authenticate_peer_cert(self):
+        """Authenticate the client per the configured client_certs.
+
+        Aborts the request w/ a 401 on failure.
+
+        """
+        cert = self.request.connection.transport.getPeerCertificate()
+        assert cert, "Expected a TLS peer cert (VERIFY_FAIL_IF_NO_PEER_CERT)"
```

Comment:
Ok, added bonus: [this assert could be disabled](https://docs.python.org/2.7/reference/simple_stmts.html#the-assert-statement). Best to do a real check.